core/src/main/java/bisq/core/dao/monitoring/ProposalStateMonitoringService.java

```diff
@@ -134,15 +134,18 @@ public void onDaoStateChanged(Block block) {
         int blockHeight = block.getHeight();
         int genesisBlockHeight = genesisTxInfo.getGenesisBlockHeight();
 
+        boolean hashChainUpdated = false;
         if (proposalStateBlockChain.isEmpty() && blockHeight > genesisBlockHeight) {
             // Takes about 150 ms for dao testnet data
             long ts = System.currentTimeMillis();
             for (int i = genesisBlockHeight; i < blockHeight; i++) {
-                maybeUpdateHashChain(i);
+                hashChainUpdated = hashChainUpdated || maybeUpdateHashChain(i);
+            }
+            if (hashChainUpdated) {
+                log.info("updateHashChain for {} items took {} ms",
```

Comment:
Perhaps could be clarified as blocks rather than items?

Reply:
Fixed in new commit.